processing/src/main/java/org/apache/carbondata/lcm/status/SegmentStatusManager.java

```diff
@@ -177,6 +178,13 @@ public ValidAndInvalidSegmentsInfo getValidAndInvalidSegments() throws IOExcepti
       }
 
     }
+
+    // remove entry in the segment index if there are invalid segments
+    if (listOfInvalidSegments.size() > 0) {
```

Comment:
segment status manager responsibility is to give list of valid and invalid segment, removing invalid segments from SegmentTaskIndexStore in SegmentStatusManager is not a good idea, I think class which is loading blocks is a better place to add this code.